lib/ex_machina/ecto.ex

```diff
@@ -82,12 +91,32 @@ defmodule ExMachina.Ecto do
     ExMachina.create(module, factory_name)
   end
 
+  defp belongs_to_assocs(%{__struct__: struct}) do
+    associations = for a <- struct.__schema__(:associations) do
+      {a, struct.__schema__(:association, a)}
+    end
+    for {a, %Ecto.Association.BelongsTo{}} <- associations, do: a
+  end
+
+  defp convert_to_changes(record) do
+    record
+    |> Map.from_struct
+    |> Map.delete(:__meta__)
+    # drop fields for `belongs_to` assocs as they cannot be handled by changeset
+    |> Map.drop(belongs_to_assocs(record))
+  end
+
   @doc """
   Saves a record using `Repo.insert!` when `create` is called.
   """
-  def save_record(module, repo, record) do
-    if repo do
-      repo.insert!(record)
-    end
+  def save_record(module, repo, %{__struct__: model} = record) do
+    struct(model)
+    # `belongs_to` assocs are not managed by changeset, add them to model
+    |> Map.merge(Map.take(record, belongs_to_assocs(record)))
```

Comment:
@paulcsmith `belongs_to` associations can be included in factory or `attrs`, they will not be included in changeset changes, however they will be retained in the resulting model.

For example, this works:
```elixir
user = create(:user)
article = create(:article, author: user)
IO.inspect article.author.name # author is retained in `create`
```
Ecto.Changeset raises error on insert if `belongs_to` association is listed in `%Ecto.Changeset{changes: %{}}`. They are however already persisted by `assoc` method.

The logic in `save_record` can be summarized as
1) Add all `belongs_to` fields to initial model
2) Create changeset from initial model, NOT changing any `belongs_to` fields
3) Insert changeset

If step 1) is skipped, example above does not work, e.g.
```elixir
IO.inspect article.author.name # raises assoc not loaded exception
```

Does this answer your question?